cleanlab/filter.py

```diff
@@ -236,50 +235,42 @@ def multiclass_crossval_predict(pyx, labels):
+def find_label_issues(labels, psx, confident_joint=None, frac_noise=1.0,
+                      num_to_remove_per_class=None, filter_by='prune_by_noise_rate',
+                      return_indices_ranked_by=None, multi_label=False, n_jobs=None, verbose=0):
+    """By default, this method returns a boolean mask for the entire dataset where True represents
+    a label issue and False represents an example that is confidently/accurately labeled.
+
+    You can return ONLY the indices of the label issues in your dataset, by setting
+    return_indices_ranked_by = {`prob_given_label`, `normalized_margin`}.
+
     number of indices returned is specified by frac_of_noise. When
     frac_of_noise = 1.0, all "confident" estimated noise indices are returned.
     * If you encounter the error 'psx is not defined', try setting n_jobs = 1.
 
+    WARNING! frac_noise and num_to_remove_per_class parameters are only supported when filter_by
+    is either 'prune_by_noise_rate', 'prune_by_class', or 'both'. They are not supported for methods
+    'confident_learning' or 'argmax_not_equal_given_label'. TODO.
+
     Parameters
     ----------
 
-    s : np.array
-      A binary vector of labels, s, which may contain mislabeling. "s" denotes
+    labels : np.array
+      A binary vector of labels, labels, which may contain mislabeling. "labels" denotes
       the noisy label instead of \\tilde(y), for ASCII encoding reasons.
 
     psx : np.array (shape (N, K))
-      P(s=k|x) is a matrix with K (noisy) probabilities for each of the N
+      P(labels=k|x) is a matrix with K (noisy) probabilities for each of the N
       examples x.
       This is the probability distribution over all K classes, for each
-      example, regarding whether the example has label s==k P(s=k|x).
+      example, regarding whether the example has label labels==k P(labels=k|x).
       psx should have been computed using 3+ fold cross-validation.
 
-    inverse_noise_matrix : np.array of shape (K, K), K = number of classes
-      A conditional probability matrix of the form P(y=k_y|s=k_s) representing
-      the estimated fraction observed examples in each class k_s, that are
-      mislabeled examples from every other class k_y. If None, the
-      inverse_noise_matrix will be computed from psx and s.
-      Assumes columns of inverse_noise_matrix sum to 1.
-
     confident_joint : np.array (shape (K, K), type int) (default: None)
-      A K,K integer matrix of count(s=k, y=k). Estimates a a confident
-      subset of the joint distribution of the noisy and true labels P_{s,y}.
+      A K,K integer matrix of count(labels=k, y=k). Estimates a a confident
+      subset of the joint distribution of the noisy and true labels P_{labels,y}.
       Each entry in the matrix contains the number of examples confidently
-      counted into every pair (s=j, y=k) classes.
+      counted into every pair (labels=j, y=k) classes.
```

Comment:
would be good to specify which function may have been previously called to produce this `confident_joint`

Reply:
good idea. done